docs/guide/usage.md

```diff
@@ -81,6 +81,45 @@ this path instead: import '~@kong/kongponents/dist/style.css' */
 </style>
 ```
 
+## SSR
+
+If you're using Kongponents in SSR mode, we advice [registering Kongponents as a Vue Plugin](#vue-plugin).
+
+Kongponents uses an external function to generate UUID strings that are mostly used in accessibility-related attributes. 
+
+In server-side rendering, these unique ids will produce hydration mismatch errors because the function returns different values on the server and client. 
+
+You can provide your own SSR-compatible function for generating random strings or ids through the Vue plugin options.
+
+
+Here's an example of how to initialize the Kongponents plugin in a [Nuxt plugin](https://nuxt.com/docs/guide/directory-structure/plugins#vue-plugins).
+
+```ts
+// plugins/kongponents.ts
+
+// Import Kongponents plugin
+import Kongponents from '@kong/kongponents'
+
+// Import Kongponents styles
+import '@kong/kongponents/dist/style.css'
+
+export default defineNuxtPlugin({
+  name: 'kongponents',
+  setup(nuxtApp) {
+    nuxtApp.vueApp.use(Kongponents, {
+      uniqueIdFn: () => useId(), // Utilize the `useId()` function from Nuxt/Vue (v3.5+)
+    })
+  },
+})
+
+Plugin options object accepts these parameters:
+
+```ts
+interface KongponentsPlugin {
+  uniqueIdFn: () => string
+}
+```
```

Comment:
docs: once we decide on a solution, we should also document that you can just call `app.provide` in the host app if you don't initialize all Kongponents as a plugin